contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -333,11 +359,24 @@ def _transcribe(item, parents, editRate, masterMobs):
                 otio.opentime.RationalTime(media_start, editRate),
                 otio.opentime.RationalTime(media_length, editRate)
             )
-            # copy the metadata from the master into the media_reference
-            media.metadata["AAF"] = mastermob_child.metadata.get("AAF", {})
+
+            # Copy the metadata from the master into the media_reference
+            clipMetadata = copy.deepcopy(mastermob_child.metadata.get("AAF", {}))
+
+            # If the composition was holding UserComments and the current masterMob has
+            # no UserComments, use the ones from the CompositionMob. But if the
+            # masterMob has any, prefer them over the compositionMob, since the
+            # masterMob is the ultimate authority for a source clip.
+            if compositionUserMetadata:
+                if "UserComments" not in clipMetadata.keys():
```

Comment:
`not in` is a check against `keys()`, don't need the extra call

```suggestion
                if "UserComments" not in clipMetadata:
```